docs/apps/srt-live-transmit.md

```diff
@@ -183,6 +185,19 @@ instead of `IP_ADD_MEMBERSHIP` and the value is set to `imr_sourceaddr` field.
 Explanations for the symbols and terms used above can be found in POSIX
 manual pages, like `ip(7)` and on Microsoft docs pages under `IPPROTO_IP`.
 
+### Medium: RTP
+
+RTP over UDP is supported for input only.
+
+With no RTP-specific URI parameters, RTP medium functions identically to UDP medium as described above.
```

Comment:
That's not exactly true because for UDP it doesnt skip first 12 bytes of the RTP header. I think at least this should be mentioned here.

And more precisely, the idea of RTP input is to extract the RTP payload part so that only this is passed to the output.

Reply:
When there are no parameters provided, RTP also does not skip the first 12 bytes since droprtpheader defaults to false. I will clarify this in the docs.

Reply:
But this doesn't make sense.

What is the actual "contents" inside the stream, the actual payload that you intend to pass through the application?

Normally, when you use UDP as input, then UDP wraps MPEG-TS. Then you pass it to another medium to the output, let's say SRT, and this way SRT passes MPEG-TS.

You can also do a "RTP over SRT", in which case you specify the medium as UDP, but the actual "payload" passed to another medium is RTP (wrapping also another format, likely MPEG-TS, but that's unimportant). Of course the reader must be aware of that and likely flip SRT to UDP, which will make the output the original RTP.

So, the medium type you specify is the medium or format expected at the input, without specifying any medium inside. This means that even if the medium is RTP, if you specify UDP as input, it means that it should read directly UDP without paying attention to what format has the payload inside. But if you specify the input as RTP, you expect exactly the RTP format, and not "anything over UDP".

The "full implementation" of the RTP input should also theoretically verify the header and parse contents to check the proper length of the header automatically, but it's ok if there is only a simplified implementation with a constant default header size, optionally to be specified by the user. But then you should skip the RTP header *always* because what you are expecting on input should always be RTP. If a user wanted no header to be skipped, they'd simply use UDP.

Reply:
Documentation updated to clarify parameter defaults via beb9d601.

Further discussion on this in https://github.com/Haivision/srt/pull/2848#pullrequestreview-1812782968.

Reply:
Apologies, your messages crossed and I missed yours.

Thank you for the above - it provided clarity and  I think I understand what you are after now and it should be addressed via [03e6cdf](https://github.com/Haivision/srt/pull/2848/commits/03e6cdfc2153c0c8c2bd336e44e68c6650399a96).

Clearly a squash will be needed at merge time :eyes:

Reply:
No squash required. Just make sure you've drawn off or merged later the latest master.